### PR TITLE
hide aggregator metric

### PR DESCRIPTION
**Pull request by WJIAN**
I want to hide some aggregations metric ,these aggregations are middle value for Post Aggs.
I modify little code to implement it.

### CODE REVIEW

**Comment by abhisant** [comment]
Thanks @WJIAN : Will check this and get back to you.

**Comment by abhisant** [comment]
@WJIAN : I didn't understand the use-case here. Could you please elaborate the significance of "hide". Will it have any change in the request and response content ?

**Comment by WJIAN** [comment]
Under some scenes，post aggregation metric is the only value I want to show in the graph, so set an option to HID
the aggregation. It doesn't change request or response, only changes graph.

**Comment by knoguchi** [comment]
Lets say you want to chart average.  Aggragtions `numerator = longsum(X)` and `denominator = longsum(Y)` then Post Aggregation `avg = numerator / denominator`.  You only want to chart `avg` but not `numerator` or `denominator`.  That's the use case.

**Comment by WJIAN** [comment]
Thank you @knoguchi , this is what I want to express. My English is very poor, please don't mind.

**Comment by nikhil2406** [comment]
@WJIAN you've added a useful feature. I have a suggestion about the way its visualized. How about adding 'hide' option as a member property (of arithmetic aggregations) instead of a checkbox?
For e.g. something similar to approxHistogram aggregation's 6 odd members, as shown in this screenshot:
screen shot 2017-07-03 at 3 51 04 pm


I think member property would help because:
- when user clicks 'hide' and 'add tag', there is no way to check if aggregation was marked as hidden or not. If we add 'hide' as a member user would see something like this: 'count: name = myaggname, type = hidden'.
- Current proposed checkbox gets displayed as a common option for all types of aggregations. Histogram type aggregations already are never shown in graph. But 'hide' option gives an impression to user that by not clicking 'hide', histogram aggregation would be displayed in graph.

**Comment by abhisant** [comment]
Will have a look into it.

File: src/partials/query.editor.html [hunk 1]
```diff
@@ -341,9 +346,8 @@
     <div class="gf-form" ng-show="ctrl.addAggregatorMode" ng-switch on="ctrl.target.currentAggregator.type">
       <input type="text" ng-switch-when="hyperUnique" class="input-small gf-form-input width-10" spellcheck='false' ng-model="ctrl.target.currentAggregator.name" placeholder="output name" ng-blur="ctrl.targetBlur()">
       <input type="text" ng-switch-when="hyperUnique" class="input-small gf-form-input width-10" spellcheck='false' ng-model="ctrl.target.currentAggregator.fieldName" ng-blur="ctrl.targetBlur()" placeholder="metric name" bs-typeahead="getMetrics" data-min-length=0>
+      <gf-form-switch ng-switch-when="hyperUnique" class="gf-form" label="Hiden" label-class="query-keyword" checked="ctrl.target.currentAggregator.hidden"></gf-form-switch>
     </div>
```
Comment (nikhil2406) [inline src/partials/query.editor.html]: this looks like a typo :)
'Hiden'

**Comment by nikhil2406** [comment]
Looks good. Thanks @WJIAN . Please correct that typo 'Hiden' and it'll good to merge.

**Comment by abhisant** [comment]
Thanks @WJIAN : Could you please also update the change in the dist.. Will merge it then.

**Pull request merged**